### PR TITLE
fix "a bytes-like object is required, not 'str'"

### PR DESCRIPTION
**Pull request by wswsmao**
### Description

I run this cmd below in python 3.11:

```bash
conda init bash
```

I get this error:

```
    Traceback (most recent call last):
      File "/usr/lib/python3.11/site-packages/conda/exception_handler.py", line 17, in __call__
        return func(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^
      File "/usr/lib/python3.11/site-packages/conda/cli/main.py", line 78, in main_subshell
        exit_code = do_call(args, parser)
                    ^^^^^^^^^^^^^^^^^^^^^
      File "/usr/lib/python3.11/site-packages/conda/cli/conda_argparse.py", line 166, in do_call
        result = getattr(module, func_name)(args, parser)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/lib/python3.11/site-packages/conda/cli/main_init.py", line 36, in execute
        return initialize(
               ^^^^^^^^^^^
      File "/usr/lib/python3.11/site-packages/conda/core/initialize.py", line 137, in initialize
        run_plan_elevated(plan2)
      File "/usr/lib/python3.11/site-packages/conda/core/initialize.py", line 853, in run_plan_elevated
        result = subprocess_call(
                 ^^^^^^^^^^^^^^^^
      File "/usr/lib/python3.11/site-packages/conda/gateways/subprocess.py", line 108, in subprocess_call
        stdout, stderr = process.communicate(input=stdin)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/lib64/python3.11/subprocess.py", line 1209, in communicate
        stdout, stderr = self._communicate(input, endtime, timeout)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/usr/lib64/python3.11/subprocess.py", line 2088, in _communicate
        input_view = memoryview(self._input)
                     ^^^^^^^^^^^^^^^^^^^^^^^
    TypeError: memoryview: a bytes-like object is required, not 'str'
```

### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Comment by wswsmao** [comment]
> We require contributors to sign our [Contributor License Agreement](https://conda.io/en/latest/contributing.html#conda-contributor-license-agreement) and we don't have one on file for @wswsmao.
> 
> In order for us to review and merge your code, please e-sign the [Contributor License Agreement PDF](https://conda.io/en/latest/contributing.html#conda-contributor-license-agreement). We then need to manually verify your signature, merge the PR ([conda/infrastructure#879](https://github.com/conda/infrastructure/pull/879)), and ping the bot to refresh the PR.

done
https://github.com/conda/infrastructure/pull/879

**Comment by beeankha** [comment]
> > We require contributors to sign our [Contributor License Agreement](https://conda.io/en/latest/contributing.html#conda-contributor-license-agreement) and we don't have one on file for @wswsmao.
> > In order for us to review and merge your code, please e-sign the [Contributor License Agreement PDF](https://conda.io/en/latest/contributing.html#conda-contributor-license-agreement). We then need to manually verify your signature, merge the PR ([conda/infrastructure#879](https://github.com/conda/infrastructure/pull/879)), and ping the bot to refresh the PR.
> 
> done [conda/infrastructure#879](https://github.com/conda/infrastructure/pull/879)

Hi @wswsmao , thanks for taking the time to create this pull request. I've looked through the signed contributor license agreements we've received recently but don't see any with your GitHub username listed; is there a chance that you might have put your given name in the "[Print Name]" _and_ "GitHub User IDs" section of the agreement instead of putting your GH username in the "GitHub User IDs" section?

**Comment by kenodegard** [comment]
@conda-bot check

File: conda/core/initialize.py [hunk 1]
```diff
@@ -852,7 +852,7 @@ def run_plan_elevated(plan):
                     rm_rf(temp_path)
 
         else:
-            stdin = json.dumps(plan, ensure_ascii=False, default=lambda x: x.__dict__)
+            stdin = json.dumps(plan, ensure_ascii=False, default=lambda x: x.__dict__).encode('utf-8')
```
Comment (kenodegard) [inline conda/core/initialize.py:855]: Minor reformat to make pre-commit/ruff happy:

```suggestion
            stdin = json.dumps(
                plan,
                ensure_ascii=False,
                default=lambda x: x.__dict__,
            ).encode('utf-8')
```

**Comment by kenodegard** [comment]
pre-commit.ci autofix

**Comment by qnm** [comment]
Confirming this change fixed the issue for me.

**Comment by jaimergp** [comment]
Isn't this a dup of https://github.com/conda/conda/pull/13240?

**Comment by wswsmao** [comment]
> > > We require contributors to sign our [Contributor License Agreement](https://conda.io/en/latest/contributing.html#conda-contributor-license-agreement) and we don't have one on file for @wswsmao.
> > > In order for us to review and merge your code, please e-sign the [Contributor License Agreement PDF](https://conda.io/en/latest/contributing.html#conda-contributor-license-agreement). We then need to manually verify your signature, merge the PR ([conda/infrastructure#879](https://github.com/conda/infrastructure/pull/879)), and ping the bot to refresh the PR.
> > 
> > 
> > done [conda/infrastructure#879](https://github.com/conda/infrastructure/pull/879)
> 
> Hi @wswsmao , thanks for taking the time to create this pull request. I've looked through the signed contributor license agreements we've received recently but don't see any with your GitHub username listed; is there a chance that you might have put your given name in the "[Print Name]" _and_ "GitHub User IDs" section of the agreement instead of putting your GH username in the "GitHub User IDs" section?

**Comment by wswsmao** [comment]
> > > We require contributors to sign our [Contributor License Agreement](https://conda.io/en/latest/contributing.html#conda-contributor-license-agreement) and we don't have one on file for @wswsmao.
> > > In order for us to review and merge your code, please e-sign the [Contributor License Agreement PDF](https://conda.io/en/latest/contributing.html#conda-contributor-license-agreement). We then need to manually verify your signature, merge the PR ([conda/infrastructure#879](https://github.com/conda/infrastructure/pull/879)), and ping the bot to refresh the PR.
> > 
> > 
> > done [conda/infrastructure#879](https://github.com/conda/infrastructure/pull/879)
> 
> Hi @wswsmao , thanks for taking the time to create this pull request. I've looked through the signed contributor license agreements we've received recently but don't see any with your GitHub username listed; is there a chance that you might have put your given name in the "[Print Name]" _and_ "GitHub User IDs" section of the agreement instead of putting your GH username in the "GitHub User IDs" section?

Hi，in this pr.
I found it has been merged
https://github.com/conda/infrastructure/pull/879

**Comment by wswsmao** [comment]
hi, @beeankha , is there any way to re-sign it?
image

**Comment by wswsmao** [comment]
Hi @kenodegard , I accidentally reopened this PR ,  is there any operations  you need to do ?

**Comment by wswsmao** [comment]
> Isn't this a dup of #13240?

I haven't found this issue before

**Pull request closed without merging**